File: adapters/python/unikernal/examples/send_example.py
import sys
import os

# Add parent directory to path to import unikernal package
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../')))

from unikernal.client import UnikernalClient

def main():
    # Initialize client
    client = UnikernalClient(service_id="python-sender")
    
    # Create a message
    message = client.create_message(
        target="node-receiver",
        intent="GREETING",
        payload={"msg": "Hello from Python!"}
    )
    
    print("Sending message via HTTP...")
    response = client.send_udl_http(message)
    
    if response:
        print("Response received:")
        print(response)

if __name__ == "__main__":
    main()

File: adapters/node/examples/sendExample.js
const { UnikernalClient } = require("../client");

async function main() {
    const client = new UnikernalClient(
        "http://localhost:4000/udl",
        "ws://localhost:4000/ws",
        "node-test-service"
    );

    const message = client.createMessage(
        "string-service",     //  use string-service
        "PING",
        {
            operation: "upper",        // upper / lower / length / concat
            text: "unikernal from node"
        }
    );

    const response = await client.sendUdlHttp(message);
    console.log("Kernel response:", response);
}

main().catch(console.error);
